packages/loader/container-loader/src/containerStateManager.ts

```diff
@@ -0,0 +1,209 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	ISequencedDocumentMessage,
+	ISnapshotTree,
+	IVersion,
+} from "@fluidframework/protocol-definitions";
+import { IGetPendingLocalStateProps, IRuntime } from "@fluidframework/container-definitions";
+import {
+	ITelemetryLoggerExt,
+	MonitoringContext,
+	PerformanceEvent,
+	UsageError,
+	createChildMonitoringContext,
+} from "@fluidframework/telemetry-utils";
+import { assert } from "@fluidframework/core-utils";
+import {
+	IDocumentStorageService,
+	IResolvedUrl,
+	ISnapshot,
+} from "@fluidframework/driver-definitions";
+import { isInstanceOfISnapshot } from "@fluidframework/driver-utils";
+import { ISerializableBlobContents, getBlobContentsFromTree } from "./containerStorageAdapter";
+import { IPendingContainerState } from "./container";
+
+export class ContainerStateManager {
```

Comment:
add some unit test for this class now that we can. this will also ensure our abstraction is easily testable which is one of the goals of the encapsulation

Reply:
I just added a couple of tests. Given that new class doesn't really do anything interesting yet, it's hard to come up with interesting tests. Next round should be some fetch snapshot tests, including fake failures to start prototyping the refresh snapshot scheduler

Reply:
To make it easily unit testable I had to move the offlineLoadEnable calculation outside of the class. I couldn't figure out how to make the sublogger to set "offlineLoadEnabled" true from the unit test file. Maybe it's simple and I can take it back inside the class.